File: app/config/i18n.py
from typing import Dict, Any
from .lang_es import lang_es
from .lang_en import lang_en

languages: Dict[str, Dict[str, Any]] = {
    "es": lang_es,
    "en": lang_en,
}

def get_text(key: str, lang: str = "es") -> str:
    if lang not in languages:
        lang = "es"

    keys = key.split(".")
    value = languages[lang]

    for k in keys:
        if isinstance(value, dict) and k in value:
            value = value[k]
        else:
            return key

    return str(value) if not isinstance(value, dict) else key

def get_translation_dict(lang: str = "es") -> Dict[str, Any]:
    if lang not in languages:
        lang = "es"
    return languages[lang]

File: frontend/src/config/i18n.ts
import { lang_es } from './lang_es';
import { lang_en } from './lang_en';

const languages = {
  es: lang_es,
  en: lang_en,
};

export type Language = keyof typeof languages;

const LANGUAGE_STORAGE_KEY = 'app_language';

export function getCurrentLanguage(): Language {
  const stored = localStorage.getItem(LANGUAGE_STORAGE_KEY) as Language;
  if (stored && stored in languages) {
    return stored;
  }
  const browserLang = navigator.language.split('-')[0];
  return (browserLang in languages ? browserLang : 'es') as Language;
}

export function setLanguage(lang: Language): void {
  localStorage.setItem(LANGUAGE_STORAGE_KEY, lang);
  window.location.reload();
}

export function getTranslations() {
  const currentLang = getCurrentLanguage();
  return languages[currentLang];
}

export { lang_es, lang_en };
